### PR TITLE
feat(typescript-sdk): implement initial version of documentation, update server types

### PR DESCRIPTION
**Pull request by fbozic**

### CODE REVIEW

File: docs/docs/client-ts-sdk.mdx [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+}
+
+export interface RpcMutateParams<Args> {
+    applicationId: ApplicationId;
+    method: string;
+    argsJson: Args;
+}
+
+export interface RpcMutateResponse<Out> {
+    output?: Out;
```
Comment (MatejVukosav) [inline docs/docs/client-ts-sdk.mdx]: Can you use the full name here `Out` -> `Output`?

File: docs/docs/client-ts-sdk.mdx [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+}
+```
+
+#### SubscriptionsClient 
+
+```typescript
+export interface SubscriptionsClient {
+    connect(connectionId?: string): void;
```
Comment (MatejVukosav) [inline docs/docs/client-ts-sdk.mdx:62]: Let's add docs in the code as well describing the types. e.g. connectionId.

Add a task in Jira and we can tackle it after Dubai if not before.

File: docs/docs/client-ts-sdk.mdx [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+    disconnect(connectionId?: string): void;
+    subscribe(applicationIds: string[], connectionId?: string): void;
+    unsubscribe(applicationIds: string[], connectionId?: string): void;
+    addCallback(callback: (data: NodeEvent) => void, connectionId?: string): void;
```
Comment (MatejVukosav) [inline docs/docs/client-ts-sdk.mdx]: Shouldn't this be `event: NodeEvent`?

File: docs/docs/client-ts-sdk.mdx [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+    title: string;
+    text: string;
+}
+
+// Create an instance of JsonRpcClient
+const rpcClient: RpcClient = new JsonRpcClient("http://localhost:2529", "/jsonrpc");
+
+// Define the parameters for the mutate
+const params: RpcMutateParams<CreatePost> = {
+    applicationId: "my-application-id",
```
Comment (MatejVukosav) [inline docs/docs/client-ts-sdk.mdx:115]: Devs could question how to obtain this `applicationId`. Can you add a short explanation?

File: docs/docs/client-ts-sdk.mdx [hunk 5]
```diff
@@ -0,0 +1,149 @@
...
+    type: 'TransactionExecuted' | 'PeerJoined'
+    data: TransactionExecuted | PeerJoined;
+}
+
+export interface TransactionExecuted {
+    hash: string;
+}
+
+export interface PeerJoined {
+    peer_id: string;
```
Comment (MatejVukosav) [inline docs/docs/client-ts-sdk.mdx]: We should update this `peer_id` to map convention -> `peerId`

File: docs/docs/client-ts-sdk.mdx (hunk 2 above)
Comment (fbozic) [reply docs/docs/client-ts-sdk.mdx:62]: I'll add a issue to GitHub as well

**Pull request merged**